Documentation/templates/MigrationGuide.md

```diff
@@ -1,23 +1,121 @@
-# Master
+# SwiftGen 6.0 Migration Guide
 
-*** Work In Progress — Will be consolidated once the release is ready ***
+If you're moving from SwiftGen 5.x to SwiftGen 6.0, you'll need to be aware that some templates have been renamed, removed or merged with others. Follow this Migration Guide to use the new name for SwiftGen 6.x of the templates you were previously using with SwiftGen 5.x.
+
+## Deprecated templates in SwiftGen 6.0
+
+### Colors
+
+| Old | New | Reason |
+| --- | --- | ------ |
+| `literals-swift3` | ✅ `literals-swift3` | |
+| `literals-swift4` | ✅ `literals-swift4` | |
+| `swift2` | ❌ _deleted_ | Really old Swift version |
+| `swift3` | ✅ `swift3` | |
+| `swift4` | ✅ `swift4` | |
+
+### Fonts
+
+| Old | New | Reason |
+| --- | --- | ------ |
+| `swift2` | ❌ _deleted_ | Really old Swift version |
+| `swift3` | ✅ `swift3` | |
+| `swift4` | ✅ `swift4` | |
+
+### Storyboards / IB
+
+The templates have been split up into separate templates for each specific functionality, in preparation of future functionalities such as accessibility identifiers.
+- One template to generate scene information.
+- One template to generate segue information.
+
+| Old | New | Reason |
+| --- | --- | ------ |
+| `swift2` | ❌ _deleted_ | Really old Swift version |
+| `swift3` | ➡️ `scenes-swift3`/`segues-swift3` | Split up into separate templates for the scenes and segues functionality |
+| `swift4` | ➡️ `scenes-swift4`/`segues-swift4` | Split up into separate templates for the scenes and segues functionality |
+
+⚠️ The `storybards` subcommand from SwiftGen 5.x has been renamed `ib` in SwiftGen 6.0, so be sure to put those templates in a `ib` subfolder and not an `storyboards` subfolder.
+
+### Strings
+
+| Old | New | Reason |
+| --- | --- | ------ |
+| `flat-swift2` | ❌ _deleted_ | Really old Swift version |
+| `flat-swift3` | ✅ `flat-swift3` | |
+| `flat-swift4` | ✅ `flat-swift4` | |
+| `structured-swift2` | ❌ _deleted_ | Really old Swift version |
+| `structured-swift3` | ✅ `structured-swift3` | |
+| `structured-swift4` | ✅ `structured-swift4` | |
+
+### XCAssets
+
+| Old | New | Reason |
+| --- | --- | ------ |
+| `swift2` | ❌ _deleted_ | Really old Swift version |
+| `swift3` | ✅ `swift3` | |
+| `swift4` | ✅ `swift4` | |
+
+## Functionality changes in SwiftGen 6.0
+
+All templates now have `swiftlint:disable all` at the top, so `swiftlint` users no longer need to ignore the generated files, although this is still highly recommended.
+
+SwiftGen 6.0 uses the latest Stencil and StencilSwiftKit libraries, so there are plenty of new features for template writers, such as variable subscripting, an `indent` filter, better error reporting, ...
+
+### Fonts
+
+The template now provides a `registerAllCustomFonts()` function, which can be useful if you use custom fonts in your Interface Builder files. Just call it when your application starts.
```

Comment:
Note that if you call this method, you don't need to list the custom fonts under the `UIAppFonts` key of your `Info.plist` anymore. Calling this method instead of listing your custom fonts in your `Info.plist` thus has two advantages: you don't have to maintain the list up-to-date anymore when you add/remove a custom font, and it also works well with custom fonts you might embed in your frameworks (which don't have that `UIAppFonts` key in their own `Info.plist`)